Compiler/Util/omc_file.h

```diff
@@ -0,0 +1,253 @@
+  }
+  if (EOF == fprintf(file->file, format, data)) {
+    ModelicaFormatError("File.writeReal: Failed to write to file: %s error: %s\n", file->name, strerror(errno));
+  }
+}
+
+#define OMC_ERROR_WRITE() ModelicaFormatError("File.writeEscape: Failed to write to file %s: %s error: %s\n", file->name, file->file, strerror(errno))
+
+static inline void om_file_write_escape(__OMC_FILE *file, const char *data, enum escape_t escape)
+{
+  if (!file->file) {
+    ModelicaFormatError("File.writeEscape: Failed to write to file: %s (not open)", file->name);
+  }
+  switch (escape) {
+  case None:
+    if (EOF == fputs(data, file->file)) {
+      ModelicaFormatError("File.writeEscape: Failed to write to file: %s error: %s\n", file->name, strerror(errno));
+    }
+    break;
+  case C:
+    while (*data) {
+      if (*data == '\n') {
+        if (fputs("\\n", file->file) < 0) OMC_ERROR_WRITE();
+      } else if (*data == '"') {
+        if (fputs("\\\"", file->file) < 0) OMC_ERROR_WRITE();
+      } else {
+        if (putc(*data, file->file) < 0) OMC_ERROR_WRITE();
+      }
+      data++;
+    }
+    break;
+  case JSON:
+    while (*data) {
+      switch (*data) {
+      case '"': if (fputs("\\\"", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\\': if (fputs("\\\\", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\n': if (fputs("\\n", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\b': if (fputs("\\b", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\f': if (fputs("\\f", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\r': if (fputs("\\r", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\t': if (fputs("\\t", file->file) < 0) OMC_ERROR_WRITE();break;
+      default:
+        if (*data < ' ') { /* Escape other control characters */
+          if (fprintf(file->file, "\\u%04x",*data) < 0) OMC_ERROR_WRITE();
+        } else {
+          if (putc(*data, file->file) < 0) OMC_ERROR_WRITE();
+        }
+      }
+      data++;
+    }
+    break;
+  case XML:
+    while (*data) {
+      switch (*data) {
+      case '<': if (fputs("&lt;", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '>': if (fputs("&gt;", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '"': if (fputs("&#34;", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '&': if (fputs("&amp;", file->file) < 0) OMC_ERROR_WRITE();break;
+      case '\'': if (fputs("&#39;", file->file) < 0) OMC_ERROR_WRITE();break;
+      default:
+        if (putc(*data, file->file) < 0) OMC_ERROR_WRITE();
+      }
+      data++;
+    }
+    break;
+  default:
+    ModelicaFormatError("File.writeEscape: No such escape enumeration: %d\n", escape);
+  }
+}
+
+static inline int om_file_seek(__OMC_FILE *file, int offset, enum whence_t whence)
+{
+  if (!file->file) {
+    return 0;
+  }
+  return 0 == fseek(file->file, offset, whence == OMC_SEEK_SET ? SEEK_SET : whence == OMC_SEEK_CURRENT ? OMC_SEEK_CURRENT : SEEK_END);
```

Comment:
shouldn't this read
... whence == OMC_SEEK_CURRENT ? **SEEK_CURRENT** : SEEK_END ... ?